api/v2/checluster_types.go

```diff
@@ -0,0 +1,637 @@
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:org.w3:link"
+	PluginRegistryURL string `json:"pluginRegistryURL"`
+	// A human readable message indicating details about why the Che deployment is in this phase.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Message"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:text"
+	Message string `json:"message,omitempty"`
+	// A brief CamelCase message indicating details about why the Che deployment is in this phase.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Reason"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:text"
+	Reason string `json:"reason,omitempty"`
+	// PostgreSQL version image is in use.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="PostgreSQL version"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:text"
+	PostgresVersion string `json:"postgresVersion,omitempty"`
+	// The resolved workspace base domain. This is either the copy of the explicitly defined property of the
+	// same name in the spec or, if it is undefined in the spec and we're running on OpenShift, the automatically
+	// resolved basedomain for routes.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Workspace base domain"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:text"
+	WorkspaceBaseDomain string `json:"workspaceBaseDomain,omitempty"`
+}
+
+// The `CheCluster` custom resource allows defining and managing Eclipse Che server installation.
+// Based on these settings, the  Operator automatically creates and maintains several ConfigMaps:
+// `che`, `plugin-registry`, `devfile-registry` that will contain the appropriate environment variables
+// of the various components of the installation. These generated ConfigMaps must NOT be updated manually.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +k8s:openapi-gen=true
+// +operator-sdk:csv:customresourcedefinitions:displayName="Eclipse Che instance Specification"
+// +operator-sdk:csv:customresourcedefinitions:order=0
+// +operator-sdk:csv:customresourcedefinitions:resources={{Ingress,v1},{Route,v1},{ConfigMap,v1},{Service,v1},{Secret,v1},{Deployment,apps/v1},{Role,v1},{RoleBinding,v1},{ClusterRole,v1},{ClusterRoleBinding,v1}}
+// +kubebuilder:storageversion
+type CheCluster struct {
```

Comment:
I might have missed it in reading this file, but consider adding `+kubebuilder:printcolumn:name` annotations here to automatically print the dashboard URL, etc. when listing objects on the cluster. For example, something like 
```
// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.cheClusterRunning",description="Status for the CheCluster"
// +kubebuilder:printcolumn:name="Dashboard URL",type="string",JSONPath=".status.cheURL",description="The URL for the Che Dashboard"
```
If this is handled by the `+operator-sdk` annotations, then ignore this comment.

Reply:
An example of how this is used in the DevWorkspace CR: [[link]](https://github.com/devfile/api/blob/d6e86811be170f92c70a15fbfea79028866448a9/pkg/apis/workspaces/v1alpha2/devworkspace_types.go#L70-L78)

Reply:
I think it is covered by +operator-sdk annotations